src/main/java/fi/metatavu/metamind/server/rest/StoriesApiImpl.java

```diff
@@ -461,57 +456,71 @@ public Response findVariable(UUID storyId, UUID variableId) {
 
   @Override
   public Response listIntents(UUID storyId) {
-    // TODO: Permission check
-    
     fi.metatavu.metamind.persistence.models.Story story = storyController.findStoryById(storyId);
     if (story == null) {
       return createBadRequest(String.format("Story %s not found", storyId)); 
     }
     
+    List<fi.metatavu.metamind.persistence.models.Intent> intents = storyController.listIntentsByStory(story);
```

Comment:
Revert these changes, all intents are now listed twice which causes unnecessary performance issues and the original code did the work correctly